bench-vortex/src/data_downloads.rs

```diff
@@ -0,0 +1,74 @@
+use crate::idempotent;
+use crate::reader::{compress_parquet_to_vortex, BATCH_SIZE};
+use arrow_array::RecordBatchReader;
+use itertools::Itertools;
+use lance::dataset::WriteParams;
+use lance::Dataset;
+use lance_parquet::arrow::arrow_reader::ParquetRecordBatchReaderBuilder as LanceParquetRecordBatchReaderBuilder;
+use parquet::arrow::arrow_reader::ParquetRecordBatchReaderBuilder;
+use std::fs::File;
+use std::path::PathBuf;
+use tokio::runtime::Runtime;
+use vortex::array::chunked::ChunkedArray;
+use vortex::array::IntoArray;
+use vortex::arrow::FromArrowType;
+use vortex::serde::WriteCtx;
+use vortex_schema::DType;
+
+pub fn download_data(fname: &str, data_url: &str) -> PathBuf {
+    idempotent(fname, |path| {
+        let mut file = File::create(path).unwrap();
+
+        reqwest::blocking::get(data_url).unwrap().copy_to(&mut file)
+    })
+    .unwrap()
+}
+
+pub fn data_lance(lance_fname: &str, read: File) -> PathBuf {
```

Comment:
This is more like parquet_to_lance right?

Reply:
Not sure about inverting the idempotent wrapper, might be useful to be able to call this functions in a non-idempotent way.